ext/gmp/gmp.c

```diff
@@ -332,12 +332,18 @@ static zend_object *gmp_clone_obj(zend_object *obj) /* {{{ */
 }
 /* }}} */
 
-static void shift_operator_helper(gmp_binary_ui_op_t op, zval *return_value, zval *op1, zval *op2) {
+static void shift_operator_helper(gmp_binary_ui_op_t op, zval *return_value, zval *op1, zval *op2, zend_uchar opcode) {
 	zend_long shift = zval_get_long(op2);
 
 	if (shift < 0) {
-		php_error_docref(NULL, E_WARNING, "Shift cannot be negative");
-		RETVAL_FALSE;
+		if (strcmp(get_active_function_name(), "main")) {
```

Comment:
I don't think this is right, what it `<<` is used inside a function/method? I wouldn't bother with it, and we don't do it for other `**` errors either.

Reply:
Ah no, it was a bad idea to differentiate `pow()` and `**` by the `get_active_function_name()`...